crossbar/router/broker.py

```diff
@@ -171,13 +171,17 @@ def _filter_publish_receivers(self, receivers, publish):
 
         # if "eligible_authid" we only accept receivers that have the correct authid
         if publish.eligible_authid:
+            eligible = set()
             for aid in publish.eligible_authid:
-                receivers = receivers & self._router._authid_to_sessions.get(aid, set())
```

Comment:
BTW, I have no idea if the list-comprehension version vs. iteratively "extend()"-ing a set is faster...;)